.github/workflows/continuous-integration.yml

```diff
@@ -0,0 +1,144 @@
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        php-version:
+          - 7.2
+
+    steps:
+      - name: "Checkout"
+        uses: actions/checkout@master
+
+      - name: "Install PHP with extensions"
+        uses: shivammathur/setup-php@master
+        with:
+          php-version: ${{ matrix.php-version }}
+          extension-csv: "intl, json, pdo"
+
+      - name: "Cache dependencies installed with composer"
+        uses: actions/cache@v1
+        with:
+          path: ~/.composer/cache
+          key: php-${{ matrix.php-version }}-composer-locked${{ github.sha }}
+          restore-keys: |
+            php-${{ matrix.php-version }}-composer-locked
+
+      - name: "Install dependencies with composer"
+        run: $(which composer) install
+
+      - name: "Lint files with parallel-lint"
+        run: vendor/bin/parallel-lint --exclude vendor .
+
+      - name: "Create build directory"
+        run: mkdir -p build
+
+      - name: "Cache phpcs cache file"
+        uses: actions/cache@v1
+        with:
+          path: ~/build/.phpcs-cache
+          key: phpcs-cache-file
+          restore-keys: |
+            phpcs-cache-file
+
+      - name: "Check code style with phpcs"
+        run: vendor/bin/phpcs --runtime-set ignore_warnings_on_exit true -p .
+
+      - name: "Run static analysis with phpstan"
+        run: vendor/bin/phpstan analyze --configuration=phpstan.neon
+
+  tests:
+    name: "Tests"
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        php-version:
+          - 7.2
+          - 7.3
+
+    steps:
+      - name: "Checkout"
+        uses: actions/checkout@master
+
+      - name: "Install PHP with extensions"
+        uses: shivammathur/setup-php@master
+        with:
+          php-version: ${{ matrix.php-version }}
+          extension-csv: "intl, json, pdo, xdebug"
+          coverage: "xdebug"
+
+      - name: "Cache dependencies installed with composer"
+        uses: actions/cache@v1
+        with:
+          path: ~/.composer/cache
+          key: php-${{ matrix.php-version }}-composer-locked-${{ github.sha }}
+          restore-keys: |
+            php-${{ matrix.php-version }}-composer-locked-
+
+      - name: "Install dependencies with composer"
+        run: $(which composer) install
+
+      - name: "Create build directory"
+        run: mkdir -p build
+
+      - name: "Copy configuration"
+        run: cp src/config.php.dist src/config.php
+
+      - name: "Dump Xdebug filter with phpunit"
+        run: vendor/bin/phpunit --dump-xdebug-filter=build/xdebug-filter.php
+
+      - name: "Run tests with phpunit"
+        run: vendor/bin/phpunit --coverage-clover=build/logs/clover.xml --coverage-text --prepend=build/xdebug-filter.php
+
+      - name: "Download code coverage uploader for Codecov.io"
+        run: curl -s https://codecov.io/bash -o codecov
+
+      - name: "Send code coverage report to Codecov.io"
+        run: bash codecov -t ${{ secrets.CODECOV_TOKEN }}
```

Comment:
The `CODECOV_TOKEN`, taken from https://codecov.io/gh/joindin/joindin-api/settings, needs to be provided as a secret in the repository settings (see https://help.github.com/en/github/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets#creating-encrypted-secrets).

Unfortunately, this has the limitation that the secrets aren't available for pull request builds (unless the author has write access).

Reply:
` CODECOV_TOKEN` has been added